unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/EmailPromptHUD/EmailPromptHUDController.cs

```diff
@@ -1,34 +1,109 @@
 ﻿using UnityEngine;
-using UnityEngine.UI;
-using TMPro;
 using DCL.Interface;
+using DCL.Helpers;
+using System.Collections;
+using DCL.Tutorial;
 
-public class EmailPromptHUDController : MonoBehaviour
+public class EmailPromptHUDController : IHUD
 {
-    public TMP_InputField inputField;
-    public GameObject cryptoUserMessage;
-    public GameObject nonCryptoUserMessage;
-    public Button sendButton;
+    const float POPUP_DELAY = 60;
 
-    void Awake()
+    EmailPromptHUDView view;
+
+    bool isPopupRoutineRunning = false;
+    Coroutine showPopupDelayedRoutine;
+
+    public EmailPromptHUDController()
+    {
+        view = UnityEngine.Object.Instantiate(Resources.Load<GameObject>("EmailPromptHUD")).GetComponent<EmailPromptHUDView>();
+        view.name = "_EmailPromptHUD";
+
+        view.OnDismiss += OnDismiss;
+        view.OnSendEmail += OnSendEmail;
+
+        view.gameObject.SetActive(false);
+    }
+
+    public void SetVisibility(bool visible)
+    {
+        if (visible)
+        {
+            Utils.UnlockCursor();
+            view.gameObject.SetActive(true);
+            view.showHideAnimator.Show();
+        }
+        else
+        {
+            view.showHideAnimator.Hide();
+        }
+    }
+
+    public void Dispose()
+    {
+        if (view != null)
+        {
+            GameObject.Destroy(view.gameObject);
+        }
+        if (showPopupDelayedRoutine != null)
+        {
+            StopPopupRoutine();
+        }
+    }
+
+    public void SetEnable(bool enable)
+    {
+        if (enable && !isPopupRoutineRunning)
+        {
+            StartPopupRoutine();
+        }
+        else if (!enable && isPopupRoutineRunning)
+        {
+            StopPopupRoutine();
+        }
+    }
+
+    void StartPopupRoutine()
+    {
+        showPopupDelayedRoutine = CoroutineStarter.Start(ShowPopupDelayed(POPUP_DELAY));
+    }
+
+    void StopPopupRoutine()
     {
-        bool hasWallet = UserProfile.GetOwnUserProfile().hasConnectedWeb3;
+        if (showPopupDelayedRoutine != null)
+        {
+            CoroutineStarter.Stop(showPopupDelayedRoutine);
+            showPopupDelayedRoutine = null;
+        }
+        isPopupRoutineRunning = false;
+    }
 
-        cryptoUserMessage.SetActive(hasWallet);
-        nonCryptoUserMessage.SetActive(!hasWallet);
+    IEnumerator ShowPopupDelayed(float seconds)
+    {
+        isPopupRoutineRunning = true;
+        yield return new WaitUntil(() => CommonScriptableObjects.rendererState.Get());
+        yield return WaitForSecondsCache.Get(seconds);
```

Comment:
What if i'm in the loading state when the time passes? Maybe we can add another `rendererState` check?

Reply:
Yes, I can add that. I didn't thought it would make much of a difference if the popup is displayed while the renderer is disabled